src/adapter/src/coord/timestamp_selection.rs

```diff
@@ -451,6 +453,32 @@ impl Coordinator {
                 &compute_instance.to_string(),
             ])
             .inc();
+        if isolation_level == &IsolationLevel::StrictSerializable && real_time_recency_ts.is_none()
+        {
+            let serializable_det = self.determine_timestamp_for(
+                self.catalog().state(),
+                session,
+                id_bundle,
+                when,
+                compute_instance,
+                timeline_context,
+                real_time_recency_ts,
+                &IsolationLevel::Serializable,
+            )?;
+            match (
+                det.timestamp_context.timestamp(),
```

Comment:
This should be moved up a bit so we don't call determine_timestamp_for again when there's no timestamp.